DMCompiler/DM/Expressions/Dereference.cs

```diff
@@ -15,6 +15,27 @@ class Dereference : LValue {
         public override DreamPath? Path => _path;
         DreamPath? _path;
 
+        public static bool DirectConvertable(DMExpression expr, DMASTDereference astNode) {
+            if (expr.Path == null && astNode.Expression is DMASTProcCall) {
+                return true;
+            }
+            else if (astNode.Expression is DMASTListIndex) {
+                return true;
+            }
+            else if (astNode.Expression is DMASTDereference deref) {
+                if (deref.Type == DMASTDereference.DereferenceType.Search) {
+                    return true;
+                }
+                if (expr is Dereference _deref) {
+                    return DirectConvertable(_deref._expr, deref);
+                }
+            }
+            else if (astNode.Expression is DMASTDereferenceProc) {
+                return true;
+            }
+            return false;
```

Comment:
A switch would be more readable, imo
```suggestion
            switch (astNode.Expression) {
                case DMASTDereference deref when deref.Type == DMASTDereference.DereferenceType.Search:
                case DMASTProcCall when expr.Path == null:
                case DMASTDereferenceProc:
                case DMASTListIndex:
                    return true;
                case DMASTDereference deref when expr is Dereference _deref:
                    return DirectConvertable(_deref._expr, deref);
                default: return false;
            }
```